openlibrary/catalog/add_book/__init__.py

```diff
@@ -646,21 +654,35 @@ def load_data(rec, account_key=None):
         del rec['cover']
     try:
         # get an OL style edition dict
-        edition = build_query(rec)
+        rec_as_edition = build_query(rec)
+        edition: dict[str, Any]
+        if existing_edition:
+            edition = build_query(existing_edition.dict()) | rec_as_edition
+            # Preserve a source_records to avoid data loss and preserve the
+            # edition's work to avoid an unnecessary database call.
+            edition['source_records'] = existing_edition.get(
+                'source_records', []
+            ) + rec.get('source_records', [])
+            edition['works'] = existing_edition.get('works', '')
```

Comment:
I think we might not need this, since `rec_as_edition` should not have a `work`, so `edition['works']` should already be equal to `existing_edition['works']`
Also not if you do need this, `works` is an array; but note if existing_edition happens to be an orphan, then we won't want this set at all to anything so that downstream code will correctly create the work.

Reply:
I removed the unnecessary setting of `edition['works']` here.